pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -710,14 +711,18 @@ private static Expression compileFunctionExpression(SqlBasicCall functionNode) {
           compilePathExpression(functionName, functionNode, path);
           return RequestUtils.getIdentifierExpression(path.toString());
         }
+        if ((functionNode.getFunctionQuantifier() != null) && ("DISTINCT".equals(
+            functionNode.getFunctionQuantifier().toString()))) {
+          functionName = "DISTINCTCOUNT";
```

Comment:
why is this `DISTINCTCOUNT` shouldn't it be derived from the functionNode? it doesn't seem like the COUNT branch fall through here.

also can we add a test?

Reply:
there is a test - the change was made because the test started failing

Reply:
yes but the test is for `count(distinct ..` rewrite, i suspect this would not work for something like `SUM(distinct ` will basically fail. and i dont see a test like that one. @xiangfu0 ^ are we planning to support anything similar to this?

Reply:
I think the issue is that the old behavior has `functionKind == COUNT`, so it goes to line 698.
The new parser makes `functionKind = OTHER_FUNCTION`.

The difference here is pinot internal has the implementation of DISTINTCOUNT as a single function, SUM(DISTINCT) cannot be treated as a single function and hence not supported.

Reply:
Ran it with some test and i think the best way to handle it is to replace with the following. 
```suggestion
          functionName = "DISTINCT" + functionName;
```
After that SUM(DISTINCT col1) at least can compile correctly but not execute, otherwise it will compute out a plan exactly the same as COUNT(DISTINCT col1) and returns a result (which clearly is not what we wanted)

Reply:
I feel should just error out here during parser to avoid extra execution overhead for sum(distinct xxx) cases